dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -113,60 +121,78 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations }: any) =>
       ...configList,
       [name]: [
         ...configList[name],
-        name === 'metrics' ? initialMetricEntry : initialDimensionEntry,
+        name === AGGREGATIONS ? initialSeriesEntry : initialDimensionEntry,
       ],
     };
     setConfigList(list);
   };
 
   const updateChart = (updatedConfigList = configList) => {
-    const qm = new QueryManager();
-    const statsTokens = qm.queryParser().parse(data.query.rawQuery).getStats();
-    const newQuery = qm
-      .queryBuilder()
-      .build(data.query.rawQuery, composeAggregations(updatedConfigList, statsTokens));
-
-    batch(async () => {
-      await handleQueryChange(newQuery);
-      await dispatch(
-        changeQuery({
-          tabId,
-          query: {
-            ...data.query,
-            [RAW_QUERY]: newQuery,
-          },
-        })
-      );
-      await fetchData();
-      await dispatch(
+    if (visualizations.vis.name === VIS_CHART_TYPES.Histogram) {
+      dispatch(
         changeVizConfig({
           tabId,
-          vizId: visualizations.vis.name,
+          vizId: curVisId,
           data: {
+            ...userConfigs,
             dataConfig: {
-              metrics: updatedConfigList.metrics,
-              dimensions: updatedConfigList.dimensions,
-              breakdowns: updatedConfigList.breakdowns,
-              span: updatedConfigList.span,
+              ...userConfigs.dataConfig,
+              [GROUPBY]: configList[GROUPBY],
```

Comment:
Try to use same format at all places.

Reply:
Hi just want to add one more support for @abasatwar's comment, please make replacements similar here for ALL visualizations not just this PR thanks!

Reply:
Hi @mengweieric ,

We have made change in this PR itself for all instances of dimensions and metrics. Once this PR is merged all the other PRs will be rebased with main to get this PRs code. Please let me know if this works. Thanks!